core/src/test/java/org/apache/iceberg/avro/TestAvroReadProjection.java

```diff
@@ -49,4 +56,27 @@
 
     return Iterables.getOnlyElement(records);
   }
+
+  @Test
+  public void testAvroArrayAsLogicalMap() throws IOException {
+    Schema writeSchema = new Schema(
+        Types.NestedField.optional(0, "map", Types.MapType.ofOptional(2, 3,
+            Types.LongType.get(),
+            Types.ListType.ofRequired(1, Types.LongType.get())
+        ))
+    );
+
+    List<Long> values1 = ImmutableList.of(101L, 102L);
+    List<Long> values2 = ImmutableList.of(201L, 202L, 203L);
+    GenericData.Record record = new GenericData.Record(AvroSchemaUtil.convert(writeSchema, "table"));
+    record.put("map", ImmutableMap.of(100L, values1, 200L, values2));
+
+    GenericData.Record projected = writeAndRead("full_projection", writeSchema, writeSchema, record);
+    Assert.assertEquals("Should contain correct value list",
+            values1,
```

Comment:
Looks like this wasn't updated.

Reply:
Oh my bad, just updated.

Reply:
Thanks! I'll commit this when tests pass.